bioio_base/reader.py

```diff
@@ -1070,6 +1083,89 @@ def get_xarray_dask_stack(self, **kwargs: Any) -> xr.DataArray:
         """
         return transforms.generate_stack(self, mode="xarray_dask_data", **kwargs)
 
+    # --- New embedded metadata properties for the manifest ---
+    def _get_dim_size(self, dim: str) -> Optional[int]:
+        if dim in self.dims.order:
+            idx = self.dims.order.index(dim)
+            return self.shape[idx]
+        return None
+
+    @property
+    @embedded_metadata(MetadataLabels.DIMENSIONS_PRESENT.value)
+    def dimensions_present(self) -> Optional[str]:
+        return self.dims.order
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_C.value)
+    def image_size_c(self) -> Optional[int]:
+        return self._get_dim_size("C")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_T.value)
+    def image_size_t(self) -> Optional[int]:
+        return self._get_dim_size("T")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_X.value)
+    def image_size_x(self) -> Optional[int]:
+        return self._get_dim_size("X")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_Y.value)
+    def image_size_y(self) -> Optional[int]:
+        return self._get_dim_size("Y")
+
+    @property
+    @embedded_metadata(MetadataLabels.IMAGE_SIZE_Z.value)
+    def image_size_z(self) -> Optional[int]:
+        return self._get_dim_size("Z")
+
+    @property
+    @embedded_metadata(MetadataLabels.PIXEL_SIZE_X.value)
+    def pixel_size_x(self) -> Optional[float]:
+        return self.physical_pixel_sizes.X
+
+    @property
+    @embedded_metadata(MetadataLabels.PIXEL_SIZE_Y.value)
+    def pixel_size_y(self) -> Optional[float]:
+        return self.physical_pixel_sizes.Y
+
+    @property
+    @embedded_metadata(MetadataLabels.PIXEL_SIZE_Z.value)
```

Comment:
I feel like this is cool, but I'm wondering if it adds more clutter than necessary compared to something like:
```
def parse_metadata(self) -> EmbeddedMetadata:
    return {
         "Pixel Size X": self.physical_pixel_sizes.X,
         ...all the others
    }
```

Reply:
I think its worth splitting them up. I was convinced by more computationally complex metadata properties that would each need helper functions. Why not just have them as properties. 

If you mean just directly referencing each property (no helper functions) I think this is alot cleaner and makes it so readers dependent on base reader dont have to reimplement or modify the parse_metadata function each time they can just add the new metadata they want.

Reply:
If they are all accessed at once via the `manifest` property does that not make the computation equivalent?